src/runtime/StateSerialization/MaybeMethodBase.cs

```diff
@@ -0,0 +1,199 @@
+        {
+            get
+            {
+                if (info == null)
+                {
+                    throw new SerializationException(DeletedMessage, innerException: deserializationException);
+                }
+                return (T)info;
+            }
+        }
+
+        public T UnsafeValue { get { return (T)info; } }
+        public string Name {get{return name;}}
+        public bool Valid => info != null;
+
+        public override string ToString()
+        {
+            return (info != null ? info.ToString() : $"missing method info: {name}");
+        }
+
+        public MaybeMethodBase(T mi)
+        {
+            info = mi;
+            name = mi?.ToString();
+            deserializationException = null;
+        }
+
+        internal MaybeMethodBase(SerializationInfo serializationInfo, StreamingContext context)
+        {
+            name = serializationInfo.GetString(SerializationName);
+            info = null;
+            deserializationException = null;
+            try
+            {
+                // Retrive the reflected type of the method;
+                var typeName = serializationInfo.GetString(SerializationType);
+                var tp = Type.GetType(typeName);
+                if (tp == null)
+                {
+                    throw new SerializationException($"The underlying type {typeName} can't be found");
+                }
+                // Get the method's parameters types
+                var field_name = serializationInfo.GetString(SerializationMethodName);
+                var param = (ParameterHelper[])serializationInfo.GetValue(SerializationParameters, typeof(ParameterHelper[]));
+                Type[] types = new Type[param.Length];
+                bool hasRefType = false;
+                for (int i = 0; i < param.Length; i++)
+                {
+                    var paramTypeName = param[i].Name;
+                    types[i] = Type.GetType(paramTypeName);
+                    if (types[i] == null)
+                    {
+                        throw new SerializationException($"The parameter of type {paramTypeName} can't be found");
+                    }
+                    else if (types[i].IsByRef)
+                    {
+                        hasRefType = true;
+                    }
+                }
+
+                MethodBase mb = null;
+                if (serializationInfo.GetBoolean(SerializationIsCtor))
+                {
+                    // We never want the static constructor.
+                    mb = tp.GetConstructor(ClassManager.BindingFlags&(~BindingFlags.Static), binder:null, types:types, modifiers:null);
+                }
+                else
+                {
+                    mb = tp.GetMethod(field_name, ClassManager.BindingFlags, binder:null, types:types, modifiers:null);
+                }
+                
+                if (mb != null && hasRefType)
+                {
+                    CheckRefTypes(mb, param);
```

Comment:
is this supposed to be `mb = CheckRefTypes(...)` ?

Reply:
Yes